### ISSUE TITLE
Missing term definitions 

### ISSUE THREAD

**Issue by storybynumbers** [codemonauts/craft-glossary#5]
I am applying the glossary to a field within a repeating matrix block, and running into this error:

```
Glossary.js?v=1655385555:formatted:1947 Uncaught TypeError: Cannot read properties of null (reading 'innerHTML')
    at content (Glossary.js?v=1655385555:formatted:1947:46)
```

Words are underlined properly, but the `#glossary-terms` element is missing a number of definitions.

**Comment by kringkaste** [comment]
Hi @storybynumbers which version of the plugin and Craft CMS do you run?

**Comment by kringkaste** [comment]
For testing, I created a matrix field with a single block with one plain text field in it. Then I added this field to a section. Add an entry in the section and added multiple blocks with the field. In my template I added the following:

```twig
{% for block in craft.entries.slug("my-first-article").one().matrix.all() %}
{{ block.contentField|glossary }}
{% endfor %}
{% hook 'glossary-terms' %}
```
And this works as expected, Did I miss something?

**Comment by storybynumbers** [comment]
Hi @kringkaste, thanks for the quick response and sorry for my slow one! 

- Craft CMS: 3.7.45.1
- Glossary: 1.0.5

I'm encountering the issue when there are multiple blocks. I have a page with three blocks, and only definitions from the last block are included in `#glossary-terms`. The terms in all three blocks are properly highlighted, but a JS error is thrown when hovering over a term in the first two blocks which is not included in the last one.

Some more context:

1. I'm including `{% hook 'glossary-terms' %}` at the entry template level
2. My entry template loops through a matrix field like so

```twig
{% include '_blocks/blocks.twig' with { blocks: entry.blocks } %}
```

This is my _blocks/blocks.twig template:

```twig
{% for block in blocks.all() %}
  {% set blockType = (block.type | kebab) %}
  {% set file = '_blocks/' ~ blockType ~ '/' ~ blockType ~ '.twig' %}
  {% include file with {block: block } %}
{% endfor %}
```

3. The mini-template where the filter is finally used looks something like this (I'm over-simplifying but this reproduces the problem for me)

```twig
{% set glossary_lang = 'glossary_' ~ craft.app.locale|slice(0,2)|lower %}
{{ block.block1Text|glossary(glossary_lang) }}
```

**Comment by storybynumbers** [comment]
Hi @kringkaste, I was able to narrow the issue down. Every time the glossary filter is applied, the list of used terms is reset. It doesn't accumulate across multiple calls. So it's not an issue with the matrix field, it's an issue with applying the glossary multiple times.

**Comment by kringkaste** [comment]
Hi @storybynumbers ! Thanks for the debugging and the PR.

**Comment by kringkaste** [comment]
Fixed in release 1.0.6 for Craft 3 and 2.0.1 for Craft 4. Thanks for debugging and providing a PR!
